app/src/main/java/cm/aptoide/pt/social/view/TimelineFragment.java

```diff
@@ -614,19 +614,22 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
   }
 
   private void handleSharePreviewAnswer() {
+
+    final ShareEvent[] share = new ShareEvent[1];
     shareDialog.cancels()
-        .doOnNext(shareEvent -> timelineAnalytics.sendShareCompleted(false))
+        .doOnNext(shareEvent -> timelineAnalytics.sendErrorShareCompleted(shareEvent, false, 0))
```

Comment:
NO ACTION REQUIRED

FYI, as you properly know already, this breaks the clean architecture rules we wanted when implementing the MVP pattern in Timeline. The timeline analytics dependency should only be used in the presenter, the presenter decides what needs to be logged or not, and we try to concentrate that logic there. 

This is not your fault since this method should have been an Observable that would emit the user response (cancel or share) to the presenter and then the presenter should log the analytics events.